### DISCUSSION TITLE
scoped subscriptions discussion

### DISCUSSION THREAD

**Discussion by JackMa1111** [pmndrs/zustand#2212] [Q&A]
I'm using Zustand to create a global store. Then, I dynamically generate a large number of component states in a loop, each with a unique id as the key. At this point, there are a large number of component states in the store.

Here's my selector:
`const selector = useCallback(
  (state: any) => {
    return state[id];
  },
  [id],
);`

Then I use useStore(selector, shallow) to listen to the component state.

I've noticed that if the state of any component changes, the selector of all components will be executed again. 
The issue I want to discuss is that, in the case of a large number of component states, this will bring corresponding performance loss. Can we associate the specific state with the state at the time of component mounting as the basis for the execution of the selector? Only when the specific state[id] changes, the corresponding selector will be executed, and then shallowEqual will be performed. 
This can avoid performance issues when a large number of components are mounted at the same time.

**Comment by dbritto-dev** [comment]
@JackMa1111 what you are trying to achieve? Do you need a global store? Wouldn't be better if you have multiple stores? I mean one store per each component
